docs/container.md

```diff
@@ -0,0 +1,119 @@
+generated in the container when starting.
+Actually, the control service is disabled.
+
+To obtain the public container image you can run:
+
+    podman pull quay.io/ovirt/ovirt-imageio
+
+## Run the container
+
+The ovirt-imageio container have an entrypoint script that takes
+additional arguments:
+
+```
+$ podman run \
+        --interactive \
+        --tty \
+        --rm \
+        quay.io/ovirt/ovirt-imageio:latest --help
+usage: entrypoint.py [-h] [--ticket-id TICKET_ID] image
+
+ovirt-imageio
+
+positional arguments:
+  image                 Path to image.
+
+options:
+  -h, --help            show this help message and exit
+  --ticket-id TICKET_ID
+                        Optional. Set the ID of the ticket.
+                        It is recommended to use with random string for better security.
+                        Defaults to the image filename.
+```
+
+To run the container and serve an image, you need to mount the
+image in the container, and pass the full path to the image
+inside the container in the command line.
+
+The container script will explore the image properties, create a ticket for
+you, serve the image using an NBD server, and expose it through http at
+the port 80 using the ovirt-imageio server.
+
+### Example session
+
+Create the image that we want to expose through the
+containerized ovirt-imageio:
+
+```
+    $ mkdir /var/tmp/images
+    $ qemu-img create -f raw /var/tmp/images/disk.raw 6g
+```
+
+Load the file ticket in the examples folder.
+
+```
+    $ podman run \
+        --interactive \
+        --tty \
+        --rm \
+        --publish 8080:80 \
+        --volume /var/tmp/images:/image:Z \
+        quay.io/ovirt/ovirt-imageio:latest --ticket-id example /image/disk.raw
+    2022-10-08 14:41:14,961 INFO    (MainThread) [server] Starting (hostname=51365cc84a8b pid=1, version=2.4.7)
+    2022-10-08 14:41:14,966 INFO    (MainThread) [services] remote.service listening on ('::', 80)
+    2022-10-08 14:41:14,966 INFO    (MainThread) [server] Initial ticket: /app/ticket.json
+    2022-10-08 14:41:14,967 INFO    (MainThread) [server] Ready for requests
+```
+
+In another shell, download the image (e.g., with `curl`):
+```
+    $ curl http://localhost:8080/images/example -o /data/tmp/download.raw
+      % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
+                                     Dload  Upload   Total   Spent    Left  Speed
+    100 6144M  100 6144M    0     0  1342M      0  0:00:04  0:00:04 --:--:-- 1347M
+```
+
+However, curl does not understand sparsified images, thus transferring
+gigabytes of zeroes through the network, and creates a fully allocated image.
+
+Using a smarter client that could skip zero extents (e.g., the example
+`imageio-client`) will result in faster download speeds, and a sparsified
+image downloaded:
+```
+$ examples/imageio-client download -f raw http://localhost:8080/images/example /data/tmp/download.raw
+[ 100% ] 6.00 GiB, 0.08 s, 79.13 GiB/s
+```
```

Comment:
It can be nice to add another example for converting the image to qcow2 format, or maybe show the default (without --format raw), that creates qcow2 image by default. We can mention that we convert the format to qcow2 since this is the best format for transferring the image to other systems.

Also I'm not sure that it is clear to users what is "sparsified image downloaded".
But we can show the result of the download with ls after the download:

    curl ...

    ls -lhs /var/tmp/downlaod.raw

    examples/imageio-client ...

    ls -lhs ls -lh /var/tmp/downlaod.qcow2

This will make it very clear why using examples/imageio-client is much better,
which raise the next issue, how users can use it? this is not part of the 
release. Can we make this another tool, part of the ovirt-imageio-client pacakge?
Or maybe another package, since it does not require the the ovirt-engine-sdk?

We can handle this later, the document is good enough even if the imageio-client
tool is only part of the source.